geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -46,19 +53,91 @@
 public class RedisHash extends AbstractRedisData {
   public static final RedisHash NULL_REDIS_HASH = new NullRedisHash();
   private HashMap<ByteArrayWrapper, ByteArrayWrapper> hash;
+  private ConcurrentHashMap<UUID, List<ByteArrayWrapper>> hScanSnapShots;
+  private ConcurrentHashMap<UUID, Long> hScanSnapShotCreationTimes;
+  private ScheduledExecutorService HSCANSnapshotExpirationExecutor = null;
+
+  private static int default_hscan_snapshots_expire_check_frequency =
+      Integer.parseInt(
+          System.getProperty("HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS",
+              "30000"));
+
+  private static int default_hscan_snapshots_milliseconds_to_live =
+      Integer.parseInt(
+          System.getProperty("HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS",
+              "30000"));
+
```

Comment:
Integer already has some shortcuts to parse system properties. I'd also suggest the properties be something like `redis.hscan-snapshot-expiry` and `redis.hscan-snapshot-cleanup-interval`. Setting them then becomes: `Integer.getInteger("redis.hscan-snapshot-expiry", 30000)`.

Reply:
ok-  will change.  thx